### PR TITLE
PYIC-7357: Remove maxlength attributes from text boxes

### PR DESCRIPTION
**Pull request by DanCorderIPV**
### What changed

Remove `maxlength` attributes from text inputs

### Why did it change

Accessibility

### Issue tracking

- [PYIC-7357](https://govukverify.atlassian.net/browse/PYIC-7357)

[PYIC-7357]: https://govukverify.atlassian.net/browse/PYIC-7357?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/kbv/fields.js [hunk 1]
```diff
@@ -38,6 +38,9 @@ module.exports = {
   [APP.QUESTION_KEYS.ITA_BANKACCOUNT]: {
     type: "text",
     validate: ["required", "numeric", { type: "exactlength", arguments: [4] }],
+    // Remove maxlength attribute that gets automatically added for exactlength validators
+    // https://github.com/HMPO/hmpo-components/blob/01473508284e1d06ebc72585f970f25f17fba49d/components/hmpo-text/macro.njk#L32
+    attributes: { "maxlength": undefined },
```
Comment (merlinc) [inline src/app/kbv/fields.js:43]: Setting things to `undefined` might look a little weird, but setting things this way in user space is the way to be overriding these properties 👍 

https://github.com/HMPO/hmpo-components/blob/master/lib/globals.js#L83-L84 shows that the `hmpoGetAttributes` will do a merge of a number of different sources of attributes, and then https://github.com/HMPO/hmpo-components/blob/master/components/hmpo-date/macro.njk#L21-L25 shows those values being overwritten where not previously defined when trying to build up the individual form element properties.

So a present key with any value should be ignored, even if it's undefined.

File: src/lib/config.js [hunk 2]
```diff
@@ -1,5 +1,11 @@
 require("dotenv").config();
 
+// These are the suffixes that the HMPO framework adds to the ID of a date
+// control to generate the IDs of the text inputs.
+const HMPO_SUFFIX_DAY = "-day";
```
Comment (merlinc) [inline src/lib/config.js:5]: Technically this is a requirement by the [Date Input field in the Design System](https://design-system.service.gov.uk/components/date-input/)

```
<div class="govuk-form-group">
          <label class="govuk-label govuk-date-input__label" for="passport-issued-day">
            Day
          </label>
          <input class="govuk-input govuk-date-input__input govuk-input--width-2" id="passport-issued-day" name="passport-issued-day" type="text" inputmode="numeric">
        </div>
```

**Review by merlinc — APPROVED** [review]
LGTM

File: tests/browser/step_definitions/enter-4-digits-bank-account-tax-credits.js [hunk 3]
```diff
@@ -19,7 +19,7 @@ When(
     const singleInputQuestionPage = new Enter4DigitsBankAccountTaxCredits(
       this.page
     );
-    await singleInputQuestionPage.answer("1234.45");
+    await singleInputQuestionPage.answer("1234");
```
Comment (blakeyp) [inline tests/browser/step_definitions/enter-4-digits-bank-account-tax-credits.js:22]: A good reverse test case proving it works 😄

**Review by blakeyp — APPROVED** [review]
Nice job cracking these 🎉
@gds-jbentham you might find this interesting

**Pull request merged**